tasmota/xsns_69_opentherm_protocol.ino

```diff
@@ -431,6 +440,26 @@ void sns_opentherm_tele_oem_diag(struct OpenThermCommandT *self)
     ResponseAppend_P(PSTR("%d"), (int)self->m_results[0].m_u16);
 }
 
+/////////////////////////////////// Boiler Boiler Lock-out Reset  //////////////////////////////////////////////////
+unsigned long sns_opentherm_send_blor(struct OpenThermCommandT *self, struct OT_BOILER_STATUS_T *status)
+{
+    AddLog(LOG_LEVEL_ERROR, PSTR("[OTH]: Call Boiler Lock-out Reset"));
+    
+    self->m_flags.notSupported = true; // Disable future calls of this command
+
+    unsigned int data = 1; //1 : “BLOR”= Boiler Lock-out Reset command
+    data <<= 8;
+    return OpenTherm::buildRequest(OpenThermMessageType::OPTH_WRITE_DATA, OpenThermMessageID::Command, data);
+}
+
+bool sns_opentherm_call_blor() 
+{
+    OpenThermCommandT *cmd = &sns_opentherm_commands[(sizeof(sns_opentherm_commands) / sizeof(OpenThermCommand))-1];
```

Comment:
here you're effectively doing `&sns_opentherm_commands[SNS_OT_COMMANDS_COUNT - 1]` thus you rely on the fact that BLOR is the last command in the list. If someone else will add another command, it will break. I would suggest doing a simple loop to find it like
```
    for (int i = 0; i < SNS_OT_COMMANDS_COUNT; ++i)
    {
        struct OpenThermCommandT *cmd = &sns_opentherm_commands[I];
        if (!strcmp(cmd->m_command_name, "BLOR") && cmd->manual) {
          cmd->m_flags.notSupported = false;
        }
    }
```

Reply:
or rename the function to something like `sns_opentherm_schedule_manual(const char* command_name)` to make it generic for one-off commands.

Reply:
SNS_OT_COMMANDS_COUNT declared below this function. I tried keep other code